packages/ember-template-compiler/lib/plugins/utils.ts

```diff
@@ -12,25 +12,41 @@ export function isStringLiteral(node: AST.Expression): node is AST.StringLiteral
   return node.type === 'StringLiteral';
 }
 
+function getLocalName(node: string | AST.Expression | AST.VarHead) {
+  if (typeof node === 'string') return node;
+
+  if (node.type === 'VarHead') {
+    return node.original;
+  }
+
+  // surely this is wrong? (does it ever occur tho?)
+  // the type of params is `| Expression`, which isn't possible for block params
+  return node;
+}
+
 export function trackLocals() {
   let locals = new Map();
 
   let node = {
-    enter(node: AST.Program | AST.Block | AST.ElementNode) {
-      for (let param of node.blockParams) {
+    enter(node: AST.Template | AST.Block | AST.ElementNode | AST.BlockStatement) {
+      let params = 'params' in node ? node.params : node.blockParams;
+      for (let param of params) {
+        let name = getLocalName(param);
```

Comment:
Once you fix the types/guards `getLocalName` shouldn't need to exist

Reply:
Still needs to exist because `param` can be of type string (from `blockParams`), which exist on `Template`